app/controllers/floating_ip_controller.rb

```diff
@@ -12,4 +12,262 @@ class FloatingIpController < ApplicationController
+    floating_ips.each do |s|
+      floating_ip = FloatingIp.find_by_id(s)
+      if floating_ip.nil?
+        add_flash(_("Floating IP no longer exists."), :error)
+      else
+        floating_ips_to_delete.push(floating_ip)
+      end
+    end
+    process_floating_ips(floating_ips_to_delete, "destroy") unless floating_ips_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "floating_ip"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Floating IP was deleted"))
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("floating_ip_edit")
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Associate Floating IP \"%{name}\"") % { :name  => @floating_ip.name },
+      :url  => "/floating_ip/edit/#{@floating_ip.id}git")
+  end
+
+  def floating_ip_form_fields
+    assert_privileges("floating_ip_edit")
+    # TODO: find_by_id_filtered fails
+    # floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    floating_ip = FloatingIp.find(params[:id])
+    network_port_ems_ref = if floating_ip.network_port
+                             floating_ip.network_port.ems_ref
+                           else
+                             ""
+                           end
+    # TODO: router field is missing!
+    # :router_id            => floating_ip.router.id,
+    render :json => {
+      :fixed_ip_address     => floating_ip.fixed_ip_address,
+      :floating_ip_address  => floating_ip.address,
+      :cloud_network_name   => floating_ip.cloud_network.name,
+      :network_port_ems_ref => network_port_ems_ref,
+      :cloud_tenant_name    => floating_ip.cloud_tenant.name,
+    }
+  end
+
+  def get_checked_floating_ip_id(params)
+    if params[:id]
+      checked_floating_ip_id = params[:id]
+    else
+      checked_floating_ips = find_checked_items
+      checked_floating_ip_id = checked_floating_ips[0] if checked_floating_ips.length == 1
+    end
+    checked_floating_ip_id
+  end
+
+  def networks_by_ems
+    assert_privileges("floating_ip_new")
+    networks = []
+    available_networks = CloudNetwork.where(:ems_id => params[:id], :external_facing => true).find_each
+    available_networks.each do |network|
+      networks << { 'name' => network.name, 'id' => network.id }
+    end
+    render :json => {
+      :available_networks => networks
+    }
+  end
+
+  def new
+    assert_privileges("floating_ip_new")
+    @floating_ip = FloatingIp.new
+    @in_a_form = true
+    @ems_choices = {}
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
+      @ems_choices[ems.name] = ems.id
+    end
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
```

Comment:
also you should turn around the indexing probably? I think the name doesn't have to be always unique. You can have tree of tenants with leaves with duplicate names.